modules/healthconnectonfhir/src/main/java/edu/stanford/healthconnectonfhir/RecordToObservationMapperImpl.kt

```diff
@@ -0,0 +1,228 @@
+package edu.stanford.healthconnectonfhir
+
+import androidx.health.connect.client.records.ActiveCaloriesBurnedRecord
+import androidx.health.connect.client.records.BloodPressureRecord
+import androidx.health.connect.client.records.BodyTemperatureRecord
+import androidx.health.connect.client.records.HeartRateRecord
+import androidx.health.connect.client.records.HeightRecord
+import androidx.health.connect.client.records.Record
+import androidx.health.connect.client.records.StepsRecord
+import androidx.health.connect.client.records.WeightRecord
+import org.hl7.fhir.r4.model.CodeableConcept
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Observation
+import org.hl7.fhir.r4.model.Period
+import org.hl7.fhir.r4.model.Quantity
+import java.util.Date
+
+class RecordToObservationMapperImpl : RecordToObservationMapper {
```

Comment:
We should mark the constructor with `@Inject constructor()` so that Hilt knows how to construct an instance. Also the class itself can be marked as internal